src/m365/entra/commands/user/user-session-revoke.spec.ts

```diff
@@ -0,0 +1,184 @@
+  });
+
+  it('fails validation if neither userId nor userName is provided', () => {
+    const actual = commandOptionsSchema.safeParse({});
+    assert.notStrictEqual(actual.success, true);
+  });
+
+  it('prompts before revoking all sign-in sessions when confirm option not passed', async () => {
+    await command.action(logger, { options: { userId: userId } });
+
+    assert(promptIssued);
+  });
+
+  it('aborts revoking all sign-in sessions when prompt not confirmed', async () => {
+    const deleteSpy = sinon.stub(request, 'delete').resolves();
+
+    await command.action(logger, { options: { userId: userId } });
+    assert(deleteSpy.notCalled);
+  });
+
+  it('revokes all sign-in sessions for a user specified by userId without prompting for confirmation', async () => {
+    const postRequestStub = sinon.stub(request, 'post').callsFake(async (opts) => {
+      if (opts.url === `https://graph.microsoft.com/v1.0/users/${userId}/revokeSignInSessions`) {
+        return;
+      }
+
+      throw 'Invalid request';
+    });
+
+    await command.action(logger, { options: { userId: userId, force: true, verbose: true } });
+    assert(postRequestStub.called);
+  });
+
+  it('revokes all sign-in sessions for a user specified by UPN while prompting for confirmation', async () => {
+    const postRequestStub = sinon.stub(request, 'post').callsFake(async (opts) => {
+      if (opts.url === `https://graph.microsoft.com/v1.0/users/${formatting.encodeQueryParameter(userName)}/revokeSignInSessions`) {
+        return;
+      }
+
+      throw 'Invalid request';
+    });
+
+    sinonUtil.restore(cli.promptForConfirmation);
+    sinon.stub(cli, 'promptForConfirmation').resolves(true);
+
+    await command.action(logger, { options: { userName: userName } });
+    assert(postRequestStub.called);
+  });
+
+  it('revokes all sign-in sessions for a user specified by UPN which starts with $ without prompting for confirmation', async () => {
+    const postRequestStub = sinon.stub(request, 'post').callsFake(async (opts) => {
+      if (opts.url === `https://graph.microsoft.com/v1.0/users('${formatting.encodeQueryParameter(userNameWithDollar)}')/revokeSignInSessions`) {
+        return;
+      }
+
+      throw 'Invalid request';
+    });
+
+    await command.action(logger, { options: { userName: userNameWithDollar, force: true, verbose: true } });
+    assert(postRequestStub.called);
+  });
+
+  it('handles error when user specified by userId was not found', async () => {
+    sinon.stub(request, 'post').callsFake(async (opts) => {
+      if (opts.url === `https://graph.microsoft.com/v1.0/users/${userId}/revokeSignInSessions`) {
+        throw {
+          error:
+          {
+            code: 'Request_ResourceNotFound',
+            message: `Resource '${userId}' does not exist or one of its queried reference-property objects are not present.`
+          }
+        };
+      }
+      throw `Invalid request`;
+    });
```

Comment:
When checking if an error is correctly processed, we don't really care whether we are doing the right request. This is validated by other tests. Here we only want to check if we correctly handle API errors.

```suggestion
    sinon.stub(request, 'post').rejects({
          error:
          {
            code: 'Request_ResourceNotFound',
            message: `Resource '${userId}' does not exist or one of its queried reference-property objects are not present.`
          }
        }
    );
```